builder/dockerfile/imagecontext.go

```diff
@@ -73,7 +75,13 @@ func (m *imageSources) Unmount() (retErr error) {
 func (m *imageSources) Add(im *imageMount) {
 	switch im.image {
 	case nil:
-		im.image = &dockerimage.Image{}
+		// set the OS for scratch images
+		os := runtime.GOOS
+		// Windows does not support scratch except for LCOW
+		if runtime.GOOS == "windows" {
+			os = "linux"
```

Comment:
I think this should have a guard for `LCOWSupported()` in this version of the codebase (although it will be removed once https://github.com/moby/moby/pull/34859 gets merged)